### ISSUE TITLE
Multi-process test run

### ISSUE THREAD

**Issue by fjarri** [fjarri/Jute.jl#12]
It would be nice to have an ability to run testcases in parallel. We have a list of tests ready, and we know all their dependencies, so it should be doable. 

One problem here is that we cannot pass functions between processes, so each process will have to perform its own test file loading. After that, tests can be identified by their path. Fixture values can be identified by their indices, so we do not have to pass them either.

The interaction of this feature with exclusive fixtures (issue #11) will, probably, be pretty awkward.

**Comment by fjarri** [comment]
With the commit 42072f88b139e8f79fcaab6d241515cf28175575, global fixtures produce exactly one value per combination of parameters, so it is now possible to tell exactly how many combinations of parameters a testcase has, and which global fixture values are being used. This makes it possible to run the same testcase with different parameter combinations on different processes, and avoid instantiating the same global fixture on different processes.
